src/message-handler/include/message-handler/IGreetHandler.h

```diff
@@ -0,0 +1,15 @@
+#ifndef __IGREETHANDLER_H_
+#define __IGREETHANDLER_H_
+
+class IGreetHandler {
```

Comment:
It was this way initially but I separated it since one is blocking and strictly using the serializer and deserializer, while the other function strictly uses the queue and is non-blocking. So the two functions did not share any members. 

Also since it has the serializer and deserializer, the message dispatcher threads would endup with a serializer. For exemple calling greet in the dispatcher thread would cause two tread to write on the stream (and the message sender writes on the stream too) causing a race condition. Seperating it in two classes removes this concern.